homeassistant/components/statistics/sensor.py

```diff
@@ -170,15 +177,27 @@ def valid_state_characteristic_configuration(config: dict[str, Any]) -> dict[str
     return config
 
 
+def valid_boundary_configuration(config: dict[str, Any]) -> dict[str, Any]:
+    """Validate that either sampling_size or max_age are provided."""
+
+    if config.get(CONF_SAMPLES_MAX_BUFFER_SIZE) is None:
+        config[CONF_SAMPLES_MAX_BUFFER_SIZE] = DEFAULT_BUFFER_SIZE
+        if config.get(CONF_MAX_AGE) is None:
```

Comment:
If `CONF_MAX_AGE` is not set, shouldn't we warn too since that's the confusing case: `CONF_MAX_AGE` does not work as expected because of the default `CONF_SAMPLES_MAX_BUFFER_SIZE`?

Reply:
Agreed